### PR TITLE
[relay] feat(comment): add handleComment, handleUpdatedComment function for S…

### PR DESCRIPTION
**Pull request by iverJisty**
…ynchornizer
x

## Summary

Add handleComment and handleUpdatedComment for Synchornizer

## Linked Issue

#128 

## Details

- Add two handler function
- Add DB schema
- Add test for synchornizer

## Impacted Areas

List the files or modules that this PR will change.

## Tests

Mention all types of tests (unit, integration, performance, etc.) that you've carried out and if any specific environment or configuration is needed to run them.

## Possible Impacts

Describe any existing features that might be affected by this change, or any potential backward compatibility issues, if any.

## Visual Materials

If your changes have UI effects, consider attaching screenshots or videos to demonstrate them.

## Verification Steps

Provide a step-by-step guide for others to reproduce your results and verify your changes.

## Todo

Break down the progress of the PR for everyone to see what else you intend to include in this PR.

-   [X] Draft implementation for DB schema
-   [X] Draft implementation for `handleComment` `handleUpdatedComment`
-   [x] Unit tests for `handleComment` `handleUpdatedComment`

## Checklist

-   [x] All new and existing tests pass
-   [x] I have updated the documentation (if applicable)
-   [x] My changes do not introduce new security vulnerabilities
-   [x] Run `yarn lint:fix`

### CODE REVIEW

File: packages/relay/src/synchornizer.ts [hunk 1]
```diff
@@ -59,30 +59,124 @@ export class UnirepSocialSynchronizer extends Synchronizer {
...
-            },
-            create: {
+        console.log('----handlePost----')
+        console.log('Epoch key:', epochKey)
+        console.log('Post id:', postId)
+        console.log('Epoch:', epoch)
+        console.log('Content:', content)
+        console.log('--------------------')
+
+        if (findPost) {
```
Comment (Lisooo790926) [inline packages/relay/src/synchornizer.ts:75]: Is this correct? The original logic is post is not found then skip the processing
Here change to post is not found, then create new one

File: packages/relay/src/synchornizer.ts [hunk 2]
```diff
@@ -59,30 +59,124 @@ export class UnirepSocialSynchronizer extends Synchronizer {
...
+        console.log('Content:', content)
+        console.log('--------------------')
+
+        const findComment = await this.db.findOne('Comment', {
+            where: {
+                transactionHash,
             },
-            update: {
+        })
+        if (findComment) {
```
Comment (Lisooo790926) [inline packages/relay/src/synchornizer.ts:123]: Need to check the same as Post. Maybe we could discuss and fix this one in further PR

File: packages/relay/src/synchornizer.ts [hunk 3]
```diff
@@ -59,30 +59,124 @@ export class UnirepSocialSynchronizer extends Synchronizer {
...
+        const commentCount = await this.db.count('Comment', {
+            postId,
+        })
+
+        db.update('Post', {
+            where: {
                 postId,
             },
+            update: {
+                commentCount: commentCount + (findComment ? 0 : 1),
```
Comment (Lisooo790926) [inline packages/relay/src/synchornizer.ts:155]: After create or update the comment, not sure this count will become correct one? 
`commentCount: commentCount`

File: packages/relay/src/synchornizer.ts [hunk 4]
```diff
@@ -77,12 +75,83 @@ export class UnirepSocialSynchronizer extends Synchronizer {
...
+
+        return true
+    }
+
+    async handleUpdatedComment({ event, db, decodedData }: EventHandlerArgs) {
+        const postId = toDecString(event.topics[2])
+        const commentId = toDecString(event.topics[3])
+        const newContent = decodedData.newContent
+
+        // FIXME: Should we check the epoch key?
```
Comment (FoodChain1028) [inline packages/relay/src/synchornizer.ts:142]: I think this should be check inside `routes`

File: packages/relay/test/synchornizer.test.ts [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+            // Check if the comment is synchronized
+            let record = await sync.db.findMany('Comment', { where: {} })
+            expect(record).to.be.not.null
+            expect(record.length).equal(1)
+            expect(record[0].commentId).equal('0')
+            expect(record[0].postId).equal('0')
+            expect(record[0].content).equal(newContent)
+        })
+    })
+})
```
Comment (FoodChain1028) [inline packages/relay/test/synchornizer.test.ts:184]: I reckon we can add a test after `epoch transition` here.

**Review by FoodChain1028 — COMMENTED** [review]
LGTM, and I think we can add an `epoch` related test here

**Review by Lisooo790926 — APPROVED** [review]

**Review by chentihe — APPROVED** [review]

**Pull request merged**